dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts

```diff
@@ -0,0 +1,101 @@
+import { getKeycloakToken } from "@e2e/utils/Auth";
+import { reader_name, reader_pw, uploader_name, uploader_pw } from "@e2e/utils/Cypress";
+import {
+  CompanyDataControllerApi,
+  Configuration,
+  DataTypeEnum,
+  LksgData,
+  LksgDataControllerApi,
+} from "@clients/backend";
+import { FixtureData } from "@e2e/fixtures/FixtureUtils";
+import { getPreparedFixture } from "@e2e/utils/GeneralApiUtils";
+import { uploadCompanyAndLksgDataViaApi } from "@e2e/utils/LksgUpload";
+import { describeIf } from "../../support/TestUtility"; // TODO use @ in path
+
+describeIf(
+  "Validates the edit button functionality on the view framework page",
+  {
+    executionEnvironments: ["developmentLocal", "ci"],
+    dataEnvironments: ["fakeFixtures", "realData"],
+  },
+  () => {
+    /**
+     * Uses the dataland API to retrieve the id of an arbitrary company with data for the specified data framework
+     *
+     * @param dataType the framework to search a company for
+     * @returns a cypress chainable containing the id of a company with data for the specified framework
+     */
+    function getFirstCompanyIdFromApiWithFramework(dataType: DataTypeEnum): Cypress.Chainable<string> {
+      return getKeycloakToken(reader_name, reader_pw).then(async (token) => {
+        const data = await new CompanyDataControllerApi(new Configuration({ accessToken: token })).getCompanies(
+          undefined,
+          new Set([dataType])
+        );
+        return data.data[0].companyId;
+      });
+    }
+
+    it("Should not display the edit and create new dataset button on the framework view page for a data reader", () => {
```

Comment:
Why is this an E2E test? Seems to me that could be a simple component test.

Reply:
Migrated to Component Test in "ViewFrameworkBase.cy.ts"